argilla-server/src/argilla_server/contexts/datasets.py

```diff
@@ -105,9 +106,21 @@ async def _touch_dataset_last_activity_at(db: AsyncSession, dataset: Dataset) ->
     )
 
 
-async def list_datasets(db: AsyncSession) -> Sequence[Dataset]:
-    result = await db.execute(select(Dataset).order_by(Dataset.inserted_at.asc()))
-    return result.scalars().all()
+async def list_datasets(db: AsyncSession, user: Optional[User] = None, **filters) -> Sequence[Dataset]:
+    """
+    List stored datasets. If `user` is provided, only datasets available to the user will be returned.
+    Additionally, filters based on `Dataset` class attributes can be applied
+
+    """
+    query = select(Dataset).filter_by(**filters)
```

Comment:
Is not possible to define the order here?

```suggestion
    query = select(Dataset).filter_by(**filters).order_by(Dataset.inserted_at.asc())
```

So you can remove the line before the return.